optuna/samplers/tpe/sampler.py

```diff
@@ -205,10 +205,9 @@ def _sample_discrete_uniform(self, distribution, below, above):
     def _sample_int(self, distribution, below, above):
         # type: (distributions.IntUniformDistribution, np.ndarray, np.ndarray) -> int
 
-        q = 1.0
-        low = distribution.low - 0.5 * q
-        high = distribution.high + 0.5 * q
-        return int(self._sample_numerical(low, high, below, above, q=q))
+        low = int(distribution.low / distribution.q)
+        high = int(distribution.high / distribution.q)
+        return int(self._sample_numerical(low, high, below, above) * distribution.q)
```

Comment:
There are two problems here.

1. You divided `low` and `high` by `q`. This means that the search space is mapped to `[low/q, high/q]`. But the spaces of values of `below` and `above` are `[low, high]`.
2. We should map the search space to `[0, (high-low)/q]` for more intuitive behavior. For example, `suggest_int(-5, 5, q=3)` suggests `-3`, `0`, or `3` in the current implementation.

If it's difficult to fix these problems, I can help you.